In [42]:
import statistics
import math

X1 = [4.00,2.67,3.65,2.11,3.21,3.60,2.80]
X2 = [3.75,2.74,3.42,1.67,3.00,3.25,2.65]
# X3 = [0.0625,0.0049,0.0529,0.1936,0.0441,0.1225,0.0225]

# X1 = [12, 5, 11, 11, 9, 18]
# X2 = [7, 3, 4, 6, 3, 13]
print(sum(X3))
X1_mean = statistics.mean(X1)
X2_mean = statistics.mean(X2)
print(f'X1_mean = {X1_mean}\nX2_mean = {X2_mean}')
D_overline = X1_mean-X2_mean
print(f'D_overline = {D_overline}')

D_list = [X1[n]-X2[n] for n in range(0, len(X1))]
print(D_list)

sum_D_list = sum(D_list)
print(f'sum_D_list = {sum_D_list}')

square_each_D_list = [item**2 for item in D_list]
print(f'square_each_D_list = {square_each_D_list}')
sum_of_square_each_D_list = sum(square_each_D_list)
print(f'sum_of_square_each_D_list = {sum_of_square_each_D_list}')

SSD = sum_of_square_each_D_list - ((sum_D_list**2)/len(X1))
print(f'SSD = {SSD}')

sample_std_dev = math.sqrt(SSD/(len(X1)-1))
print(f'sample_std_dev = {sample_std_dev}')

est_std_error = sample_std_dev/math.sqrt(len(X1))
print(f'est_std_error = {est_std_error}')

t = (D_overline-0)/est_std_error
print(f't = {t}')

d = D_overline/sample_std_dev
print(f'd = {d}')

0.503
X1_mean = 3.1485714285714286
X2_mean = 2.9257142857142857
D_overline = 0.22285714285714286
[0.25, -0.07000000000000028, 0.22999999999999998, 0.43999999999999995, 0.20999999999999996, 0.3500000000000001, 0.1499999999999999]
sum_D_list = 1.5599999999999996
square_each_D_list = [0.0625, 0.00490000000000004, 0.05289999999999999, 0.19359999999999997, 0.044099999999999986, 0.12250000000000007, 0.022499999999999975]
sum_of_square_each_D_list = 0.503
SSD = 0.15534285714285728
sample_std_dev = 0.16090517763725384
est_std_error = 0.06081644067012074
t = 3.664422652847441
d = 1.3850215768665575


#### Markdown
$\overline{D}=0.22285714285714286$

$\Sigma{D}=1.5599999999999996$

$\Sigma{D^2}=0.503$

$SS_D=0.15534285714285728$

$S_D=0.16090517763725384$

$S_\overline{D}=0.06081644067012074$

$t=3.664422652847441$

$d=1.3850215768665575$

### Sample Sums of Squares
$SS_D=\Sigma{D}^2-\dfrac{(\Sigma{D})^2}{n}$

#### SAMPLE STANDARD DEVIATION, $S_D$

$S_D = \sqrt{\dfrac{SS_D}{n-1}}$

#### ESTIMATED STANDARD ERROR, $S_\overline{D}$

$S_\overline{D}=\dfrac{S_D}{\sqrt{n}}$

$t = \dfrac{\overline{D}-\mu_{D_{hyp}}}{S_\overline{D}}$

$d=\dfrac{\overline{D}}{S_D}$